src/wp-includes/user.php

```diff
@@ -2112,10 +2112,16 @@ function wp_insert_user( $userdata ) {
 		return new WP_Error( 'user_login_too_long', __( 'Username may not be longer than 60 characters.' ) );
 	}
 
+	// Username must be unique.
 	if ( ! $update && username_exists( $user_login ) ) {
 		return new WP_Error( 'existing_user_login', __( 'Sorry, that username already exists!' ) );
 	}
 
+	// Username must not exist as an existing email.
+	if ( email_exists( $user_login ) ) {
+		return new WP_Error( 'existing_user_login_as_email', 'Sorry, that username is not available.' );
```

Comment:
Instead of a static string, use the translation function here.

```suggestion
		return new WP_Error( 'existing_user_login_as_email', __( 'Sorry, that username is not available.' ) );
```